plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/internal/actions/DefaultActionDispatcher.java

```diff
@@ -87,11 +87,23 @@ public class DefaultActionDispatcher extends Disposable implements ActionDispatc
    protected Provider<GLSPClient> client;
 
    public DefaultActionDispatcher() {
+      this.initialize();
+   }
+
+   protected void initialize() {
       this.name = getClass().getSimpleName() + " " + COUNT.incrementAndGet();
-      this.thread = new Thread(this::runThread);
-      this.thread.setName(this.name);
-      this.thread.setDaemon(true);
-      this.thread.start();
+      this.thread = this.createThread();
+      this.initializeThread(thread, name);
+      thread.start();
+   }
+
+   protected Thread createThread() {
+      return new Thread(this::runThread);
```

Comment:
`runThread` and  `handleNextAction` is private. Would be easier to override if they are visible.

Reply:
Covered by https://github.com/eclipse-glsp/glsp-server/pull/236